src/main/java/org/testeditor/fixture/host/net/Connection.java

```diff
@@ -0,0 +1,225 @@
+     *
+     * @param s3270Path
+     *            the path to your x3270 installation of the s3270 or ws3270.exe
+     *            application.
+     * @param hostname
+     *            the name of the host to connect to
+     * @param port
+     *            the port of the mainframe to connnect to
+     * @param type
+     *            the type e.g 3278 or 23279 see {@link TerminalType}
+     * @param mode
+     *            this is number from 2 to 5 see {@link TerminalMode}
+     *
+     */
+    public Connection connect(String s3270Path, String hostname, int port, TerminalType type, TerminalMode mode,
+            CharacterSet charSet) {
+        this.hostname = hostname;
+        String commandLine = String.format("%s -charset %s -model %s-%d %s:%d -utf8", s3270Path, charSet.getCharSet(),
+                type.getType(), mode.getMode(), hostname, port);
+        try {
+            logger.debug("executing " + commandLine);
+            s3270Process = Runtime.getRuntime().exec(commandLine);
+
+            out = new PrintWriter(new OutputStreamWriter(s3270Process.getOutputStream()));
+            in = new BufferedReader(new InputStreamReader(s3270Process.getInputStream()));
+
+            return this;
+
+        } catch (final IOException ex) {
+            throw new RuntimeException("IO Exception while starting " + s3270Path, ex);
+        }
+
+    }
+
+    /**
+     * Disconnect from mainframe. Destroys all opened Input- and OutputStreams.
+     */
+    public boolean disconnect() {
+        processAvailable();
+        doQuit();
+        stop2370Process();
+        try {
+            s3270Process.waitFor(5, TimeUnit.SECONDS);
+        } catch (final InterruptedException ex) {
+            logger.error("Something went wrong during termination of s2370 process");
+        }
+        try {
+            in.close();
+        } catch (final IOException ex) {
+            logger.error("Something went wrong during closing InputStreamreader of s2370 process");
+        }
+        cleanup();
+        logger.info("Disconnected successfully from host : {} ", hostname);
+        return isConnected();
+    }
+
+    /**
+     * reset all opened streams
+     */
+    private void cleanup() {
+        out.close();
+        in = null;
+        out = null;
+        s3270Process = null;
+    }
+
+    /**
+     * perform a quit statement
+     */
+    private void doQuit() {
+        out.println("quit");
+        out.flush();
+    }
+
+    /**
+     * stop the s3270 process
+     */
+    private void stop2370Process() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Thread.sleep(1000);
+                    if (s3270Process != null) {
+                        s3270Process.destroy();
+                    }
+                } catch (final InterruptedException ex) {
+                    if (s3270Process != null) {
+                        s3270Process.destroy();
```

Comment:
duplicate here. The `InterruptedException` is only for the sleeping so it should wrap just the `Thread.sleep(...)`